### ISSUE TITLE
feat(linter): eslint/no-irregular-whitespace

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#612]
Implement ESLint's [`no-irregular-whitespace`](https://eslint.org/docs/latest/rules/no-irregular-whitespace#rule-details) rule.

To get started, run
```sh
just new-rule no-irregular-whitespace
```

_note: this is not a sale error_

**Comment by koddsson** [comment]
Yo! I don't really Rust but I'm messing around trying to implement this rule. Is it correct of me to think that the whitespaces don't show up as a AST node? Should I get the source code from the context instead and grep that for a irregular whitespace and add a diagnostic?

**Comment by koddsson** [comment]
[Looks like that's how ESlint does it :)](https://github.com/eslint/eslint/blob/main/lib/rules/no-irregular-whitespace.js)

**Comment by Boshen** [comment]
We have the parser, which means we can store the irregular whitespaces into a data structure (e.g. [Trivias](https://github.com/web-infra-dev/oxc/blob/main/crates/oxc_parser/src/lexer/trivia_builder.rs) in our case)

And then print them out from the linter.

**Comment by DeividAlmeida** [comment]
it's done!!
